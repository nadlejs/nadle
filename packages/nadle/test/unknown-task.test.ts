import { it, describe } from "vitest";
import { createExec, expectFail } from "setup";

describe("when passing unknown tasks", () => {
	const exec = createExec({ config: "unknown" });

	it("should throw error without any suggestions if not find any similar tasks", async () => {
		await expectFail(() => exec`unknown`);
	});

	it("should throw error with suggestions if find similar tasks", async () => {
		await expectFail(() => exec`boot`);
		await expectFail(() => exec`build`);
		await expectFail(() => exec`compile`);
	});
});
