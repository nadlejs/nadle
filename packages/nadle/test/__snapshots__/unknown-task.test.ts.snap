// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`when passing unknown tasks > should throw error with suggestions if find similar tasks > stderr 1`] = `[error] Task <Yellow><Bold>boot</BoldDim></Yellow> not found.`;

exports[`when passing unknown tasks > should throw error with suggestions if find similar tasks > stderr 2`] = `[error] Task <Yellow><Bold>build</BoldDim></Yellow> not found. Did you mean <Yellow><Bold>buildDev</BoldDim></Yellow>?`;

exports[`when passing unknown tasks > should throw error with suggestions if find similar tasks > stderr 3`] = `[error] Task <Yellow><Bold>compile</BoldDim></Yellow> not found. Did you mean <Yellow><Bold>compileTs</BoldDim></Yellow>, <Yellow><Bold>compileJs</BoldDim></Yellow>, or <Yellow><Bold>compileCss</BoldDim></Yellow>?`;

exports[`when passing unknown tasks > should throw error with suggestions if find similar tasks > stdout 1`] = `
[log] <Bold><Cyan>üõ†Ô∏è Welcome to Nadle {version}!</Cyan></BoldDim>
[log] Using Nadle from /ROOT/lib/index.js
[log] Loaded configuration from /ROOT/test/__fixtures__/main/nadle.unknown.ts

[log]
<Bold><Red>RUN FAILED</Red></BoldDim> in <Bold>{duration}</BoldDim> <Dim>(<Bold>0</BoldDim><Dim> task executed, <Bold>0</BoldDim><Dim> task failed)</BoldDim>
[log]
For more details, re-run the command with the <Yellow>--stacktrace</Yellow> option to display the full error and help identify the root cause.
`;

exports[`when passing unknown tasks > should throw error with suggestions if find similar tasks > stdout 2`] = `
[log] <Bold><Cyan>üõ†Ô∏è Welcome to Nadle {version}!</Cyan></BoldDim>
[log] Using Nadle from /ROOT/lib/index.js
[log] Loaded configuration from /ROOT/test/__fixtures__/main/nadle.unknown.ts

[log]
<Bold><Red>RUN FAILED</Red></BoldDim> in <Bold>{duration}</BoldDim> <Dim>(<Bold>0</BoldDim><Dim> task executed, <Bold>0</BoldDim><Dim> task failed)</BoldDim>
[log]
For more details, re-run the command with the <Yellow>--stacktrace</Yellow> option to display the full error and help identify the root cause.
`;

exports[`when passing unknown tasks > should throw error with suggestions if find similar tasks > stdout 3`] = `
[log] <Bold><Cyan>üõ†Ô∏è Welcome to Nadle {version}!</Cyan></BoldDim>
[log] Using Nadle from /ROOT/lib/index.js
[log] Loaded configuration from /ROOT/test/__fixtures__/main/nadle.unknown.ts

[log]
<Bold><Red>RUN FAILED</Red></BoldDim> in <Bold>{duration}</BoldDim> <Dim>(<Bold>0</BoldDim><Dim> task executed, <Bold>0</BoldDim><Dim> task failed)</BoldDim>
[log]
For more details, re-run the command with the <Yellow>--stacktrace</Yellow> option to display the full error and help identify the root cause.
`;

exports[`when passing unknown tasks > should throw error without any suggestions if not find any similar tasks > stderr 1`] = `[error] Task <Yellow><Bold>unknown</BoldDim></Yellow> not found.`;

exports[`when passing unknown tasks > should throw error without any suggestions if not find any similar tasks > stdout 1`] = `
[log] <Bold><Cyan>üõ†Ô∏è Welcome to Nadle {version}!</Cyan></BoldDim>
[log] Using Nadle from /ROOT/lib/index.js
[log] Loaded configuration from /ROOT/test/__fixtures__/main/nadle.unknown.ts

[log]
<Bold><Red>RUN FAILED</Red></BoldDim> in <Bold>{duration}</BoldDim> <Dim>(<Bold>0</BoldDim><Dim> task executed, <Bold>0</BoldDim><Dim> task failed)</BoldDim>
[log]
For more details, re-run the command with the <Yellow>--stacktrace</Yellow> option to display the full error and help identify the root cause.
`;
